src/libmongoc/src/mongoc/mongoc-uri.c

```diff
@@ -1621,27 +1620,9 @@ mongoc_uri_parse (mongoc_uri_t *uri, const char *str, bson_error_t *error)
       }
    }
 
-   /* TODO (CDRIVER-3723) Consider moving all "finalize" calls into one function
-    * that is additionally called after initial SRV and TXT records are applied
-    * in mongoc_topology_new. */
-   if (!mongoc_uri_finalize_tls (uri, error)) {
-      goto error;
-   }
-
-   require_auth = uri->username != NULL;
-   if (!mongoc_uri_finalize_auth (uri, error, require_auth)) {
-      goto error;
-   }
-
-   if (!mongoc_uri_finalize_directconnection (uri, error)) {
-      goto error;
-   }
-
-   if (!mongoc_uri_finalize_loadbalanced (uri, error)) {
-      goto error;
-   }
-
-   if (!mongoc_uri_finalize_srv (uri, error)) {
+   /* TODO (CDRIVER-3723) Consider calling mongoc_uri_finalize_options after 
```

Comment:
Can this update also be applied to [`mongoc_topology_new()`](https://github.com/mongodb/mongo-c-driver/blob/a203bafab7d412ed05f33b99229bacbc043c2268/src/libmongoc/src/mongoc/mongoc-topology.c#L515) to resolve CDRIVER-3723 and address/remove this comment?
```diff
-   if (!mongoc_uri_finalize_loadbalanced (topology->uri,
-                                          &topology->scanner->error)) {
-      topology->valid = false;
-   }
-
-   if (!mongoc_uri_finalize_srv (topology->uri, &topology->scanner->error)) {
-      topology->valid = false;
-   }
+   if (!mongoc_uri_finalize_options (topology->uri,
+                                     &topology->scanner->error)) {
+      topology->valid = false;
+   }
```

Reply:
I considered that, but erred on the side of minimizing changes for this PR. Two concerns:

 * Swapping to `mongoc_uri_finalize_options` could technically change behavior. The `loadbalanced` and `srv` functions are read-only and take a `const mongoc_uri_t*` but `mongoc_uri_finalize_options` could possibly modify the struct via the `tls` and `auth` functions.
 * [CDRIVER-3723](https://jira.mongodb.org/browse/CDRIVER-3723) is still going to require more comprehensive changes, so this would be incomplete.